announcement/announcement-tool/tool/src/java/org/sakaiproject/announcement/tool/AnnouncementAction.java

```diff
@@ -466,6 +469,10 @@ static String getAnnouncementRange(AnnouncementMessage a)
 		{
 			return rb.getString("gen.public");
 		}
+		else if (a.getProperties().getProperty("selectedRoles") != null)
+		{
+			return a.getProperties().getProperty("selectedRoles").replaceAll("[\\[\\]]", "");
```

Comment:
Define a constant for this.

private static final String SELECTED_ROLES_PROPERTY = "selectedRoles";

Reply:
Why do you do this replaceAll?